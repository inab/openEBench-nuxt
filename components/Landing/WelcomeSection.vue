<template>
	<v-container>
		<v-row column align="end">
			<v-col align="left" sm="12" offset-md="0" md="7">
				<h1 class="text-h3 mb-4 primary--text">Welcome to OpenEBench</h1>
				<h2 class="text-subtitle-1 mb-5">
					The ELIXIR gateway to benchmarking communities, software monitoring,
					and quality metrics for life sciences tools and workflows.
				</h2>
				<v-row class="mt-0 mt-lg-10" align="end">
					<v-col align="center">
						<v-avatar color="primary" size="62" class="mb-10">
							<v-icon dark>$vuetify.icons.values.benchmark</v-icon>
						</v-avatar>
						<p class="body-1">
							I want to search for benchmarking communities and results
						</p>
						<v-btn color="primary" class="mt-10" large block to="/benchmarking">
							Scientific Benchmarking
						</v-btn>
					</v-col>
					<v-col align="center" class="mt-10 mt-sm-0">
						<v-avatar color="accent" size="62" class="mb-10">
							<v-icon dark>$vuetify.icons.values.observatory</v-icon>
						</v-avatar>
						<p class="body-1">
							I want to see how to develop sustainable research tools
						</p>
						<v-btn color="accent" class="mt-10" large block to="/observatory">
							Tools Observatory
						</v-btn>
					</v-col>
				</v-row>
			</v-col>
			<v-col v-if="$vuetify.breakpoint.mdAndUp" md="5">
				<v-img :src="illustration" contain max-height="350" />
			</v-col>
		</v-row>
	</v-container>
</template>

<script>
export default {
	name: 'WelcomeSection',
	data() {
		return {
			illustration: require('~/static/images/illustrations/chart-person.svg'),
		};
	},
};
</script>
