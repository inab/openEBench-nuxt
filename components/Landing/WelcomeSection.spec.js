import { mount } from '@vue/test-utils';
import WelcomeSection from './WelcomeSection.vue';

const factory = () => {
	return mount(WelcomeSection, {
		...createComponentMocks({}),
		mocks: {
			$vuetify: { breakpoint: {} },
		},
	});
};

describe('WelcomeSection', () => {
	it('is instantiated', () => {
		const wrapper = factory();
		expect(wrapper).toBeTruthy();
	});

	it('should match snapshot', () => {
		const wrapper = factory();
		expect(wrapper.html()).toMatchSnapshot();
	});

	it('contains illustration image on large devices and hides it on small devices', async () => {
		const wrapper = factory();

		wrapper.vm.$vuetify.breakpoint.mdAndUp = true;
		await wrapper.vm.$nextTick();
		expect(wrapper.find('.v-image').exists()).toBe(true);

		wrapper.vm.$vuetify.breakpoint.mdAndUp = false;
		await wrapper.vm.$nextTick();
		expect(wrapper.find('.v-image').exists()).toBe(false);
	});
});
