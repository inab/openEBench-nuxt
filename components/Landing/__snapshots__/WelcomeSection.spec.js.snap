// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeSection should match snapshot 1`] = `
"<div class=\\"container\\">
  <div column=\\"\\" class=\\"row align-end\\">
    <div align=\\"left\\" class=\\"col-sm-12 col-md-7 offset-md-0 col\\">
      <h1 class=\\"text-h3 mb-4 primary--text\\">Welcome to OpenEBench</h1>
      <h2 class=\\"text-subtitle-1 mb-5\\">
        The ELIXIR gateway to benchmarking communities, software monitoring,
        and quality metrics for life sciences tools and workflows.
      </h2>
      <div class=\\"row mt-0 mt-lg-10 align-end\\">
        <div align=\\"center\\" class=\\"col\\">
          <div class=\\"v-avatar mb-10 primary\\" style=\\"height: 62px; min-width: 62px; width: 62px;\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate material-icons theme--dark\\">$vuetify.icons.values.benchmark</i></div>
          <p class=\\"body-1\\">
            I want to search for benchmarking communities and results
          </p> <a class=\\"mt-10 v-btn v-btn--block v-btn--is-elevated v-btn--has-bg v-btn--router theme--light v-size--large primary\\"><span class=\\"v-btn__content\\">
						Scientific Benchmarking
					</span></a>
        </div>
        <div align=\\"center\\" class=\\"mt-10 mt-sm-0 col\\">
          <div class=\\"v-avatar mb-10 accent\\" style=\\"height: 62px; min-width: 62px; width: 62px;\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate material-icons theme--dark\\">$vuetify.icons.values.observatory</i></div>
          <p class=\\"body-1\\">
            I want to see how to develop sustainable research tools
          </p> <a class=\\"mt-10 v-btn v-btn--block v-btn--is-elevated v-btn--has-bg v-btn--router theme--light v-size--large accent\\"><span class=\\"v-btn__content\\">
						Tools Observatory
					</span></a>
        </div>
      </div>
    </div>
    <div class=\\"col-md-5 col\\">
      <div class=\\"v-image v-responsive theme--light\\" style=\\"max-height: 350px;\\">
        <div class=\\"v-responsive__content\\"></div>
      </div>
    </div>
  </div>
</div>"
`;
